ui/src/api/googleOauth.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  API_CLIENT_ID,
+  API_CLIENT_SECRET,
+  GOOGLE_CLIENT_ID,
+  REDIRECT_URL,
+  SERVER_URL,
+} from '../globals/constants';
+import {setToken} from '../utils/token';
+import {NavigateFunction} from 'react-router-dom';
+
+export const googleOauth = async () => {
+  window.location.href =
+    'https://accounts.google.com/o/oauth2/v2/auth?' +
+    'scope=https%3A//www.googleapis.com/auth/userinfo.email%20' +
+    'https%3A//www.googleapis.com/auth/userinfo.profile&' +
+    'include_granted_scopes=true&' +
+    'response_type=token&' +
+    'state=state_parameter_passthrough_value&' +
+    `redirect_uri=${REDIRECT_URL}&` +
+    `client_id=${GOOGLE_CLIENT_ID}`;
+};
+
+export const sendGoogleAccessToken = async (
```

Comment:
When I read this function name, I have many questions:
1. Who are we sending the google access token to?
2. Why are we sending google access token?

Let's rename this so these two questions are answered by the name:
Something like: `convertAccessToken` and have a JSDoc to explain what's happening here i.e. exchanging google's access_token for our API's access token.